_studio/shared/umc/codec/mpeg2_dec/src/umc_mpeg2_decoder.cpp

```diff
@@ -903,6 +903,32 @@ namespace UMC_MPEG2_DECODER
         return false;
     }
 
+    void MPEG2Decoder::EliminateSliceErrors(MPEG2DecoderFrame& frame, uint8_t fieldIndex)
+    {
+        MPEG2DecoderFrameInfo & frameInfo = *frame.GetAU(fieldIndex);
+        size_t sliceCount = frameInfo.GetSliceCount();
+
+        for (size_t sliceNum = 0; sliceNum < sliceCount; sliceNum++)
+        {
+            auto slice = frameInfo.GetSlice(sliceNum);
+            auto sliceHeader = slice->GetSliceHeader();
+
+            // Slice may cover just a part of a row (not full row), this is indicated by macroblockAddressIncrement
+            // Looping over slices, we heed to check if next slice has macroblockAddressIncrement > 0,
+            // this means that we need to recalculate numberMBsInSlice for previous slice
+            if (sliceNum > 0 && sliceHeader.macroblockAddressIncrement > 0)
+            {
+                auto prevSlice = frameInfo.GetSlice(sliceNum - 1); //take previous
+                auto& prevSliceHeader = prevSlice->GetSliceHeader();
```

Comment:
Added "&" in new patch set. Previously, without it, a copy of prevSliceHeader was created, but we need to change prevSliceHeader.numberMBsInSlice value in line 926. So now it is changing correctly.